app/components-react/sidebar/AppsNav.tsx

```diff
@@ -0,0 +1,153 @@
+import { useVuex } from '../hooks';
+
+/**
+ * The default amount the nav bar should scroll when clicking the scroll arrow buttons.
+ */
+const DEFAULT_SCROLL_DELTA = 43;
+
+export default function AppsNav() {
+  const { PlatformAppsService, NavigationService } = Services;
+
+  const { currentPage, navApps, selectedApp } = useVuex(() => ({
+    currentPage: NavigationService.state.currentPage,
+    navApps: PlatformAppsService.views.enabledApps.filter(app => {
+      return !!app.manifest.pages.find(page => {
+        return page.slot === EAppPageSlot.TopNav;
+      });
+    }),
+    selectedApp: NavigationService.state.params.appId,
+  }));
+
+  const [upArrowVisible, setUpArrowVisible] = useState(false);
+  const [downArrowVisible, setDownArrowVisible] = useState(false);
+
+  const scroll = useRef<HTMLDivElement>(null);
+
+  const resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+    entries.forEach(entry => {
+      if (entry.target.clientHeight) {
+        handleScroll();
+      }
+    });
+  });
+
+  useEffect(handleScroll, [navApps]);
+
+  useEffect(() => {
+    if (!scroll.current) return;
+    resizeObserver.observe(scroll.current);
+
+    return () => {
+      if (!scroll.current) return;
+      resizeObserver.unobserve(scroll.current);
+    };
+  }, [scroll.current]);
+
+  function isSelectedApp(appId: string) {
+    return currentPage === 'PlatformAppMainPage' && selectedApp === appId;
+  }
+
+  function isPopOutAllowed(app: ILoadedApp) {
+    const topNavPage = app.manifest.pages.find(page => page.slot === EAppPageSlot.TopNav);
+    if (!topNavPage) return false;
+
+    // Default result is true
+    return topNavPage.allowPopout == null ? true : topNavPage.allowPopout;
+  }
+
+  function popOut(app: ILoadedApp) {
+    if (!isPopOutAllowed(app)) return;
+    PlatformAppsService.actions.popOutAppPage(app.id, EAppPageSlot.TopNav);
+  }
+
+  function refreshApp(appId: string) {
+    PlatformAppsService.actions.refreshApp(appId);
+  }
+
+  function navigateApp(appId: string) {
+    NavigationService.actions.navigate('PlatformAppMainPage', { appId });
+  }
+
+  function iconSrc(appId: string, path: string) {
+    return PlatformAppsService.views.getAssetUrl(appId, path) || undefined;
+  }
+
+  function scrollUp() {
+    scrollNav(-DEFAULT_SCROLL_DELTA);
+  }
+
+  function scrollDown() {
+    scrollNav(DEFAULT_SCROLL_DELTA);
+  }
+
+  function handleScroll() {
+    const el = scroll.current;
+    if (!el) return;
+    if (el.scrollTop > 0) {
+      setUpArrowVisible(true);
+    } else {
+      setUpArrowVisible(false);
+    }
+    if (el.scrollHeight - el.scrollTop === el.clientHeight) {
+      setDownArrowVisible(false);
+    } else if (el.scrollHeight > el.clientHeight) {
+      setDownArrowVisible(true);
+    }
+  }
+
+  function scrollNav(vertical: number) {
+    scroll.current && scroll.current.scrollBy({ top: vertical, behavior: 'smooth' });
+  }
+
+  function refreshIcon(app: ILoadedApp) {
```

Comment:
I'm suggesting renaming it to `renderRefreshIcon()` or use `RefreshIcon` component name. The `refreshIcon` naming looks like some action